drivers/gpio/gpio_rpi_pico.c

```diff
@@ -170,23 +170,28 @@ static int gpio_rpi_bank_init(const struct device *dev)
 	return 0;
 }
 
-#define GPIO_RPI_INIT(idx)							\
-	static void bank_##idx##_config_func(void)				\
-	{									\
-		IRQ_CONNECT(DT_INST_IRQN(idx), DT_INST_IRQ(idx, priority),	\
-			    gpio_rpi_isr, DEVICE_DT_INST_GET(idx), 0);		\
-		irq_enable(DT_INST_IRQN(idx));					\
-	}									\
-	static const struct gpio_rpi_config gpio_rpi_##idx##_config = {		\
-		.bank_config_func = bank_##idx##_config_func,			\
-	};									\
-										\
-	static struct gpio_rpi_data gpio_rpi_##idx##_data;			\
-										\
-	DEVICE_DT_INST_DEFINE(idx, gpio_rpi_bank_init, NULL,			\
-				&gpio_rpi_##idx##_data,				\
-				&gpio_rpi_##idx##_config,			\
-				POST_KERNEL, CONFIG_GPIO_INIT_PRIORITY,		\
+#define GPIO_RPI_INIT(idx)									\
+	static void bank_##idx##_config_func(void)						\
+	{											\
+		IRQ_CONNECT(DT_INST_IRQN(idx), DT_INST_IRQ(idx, priority),			\
+			    gpio_rpi_isr, DEVICE_DT_INST_GET(idx), 0);				\
+		irq_enable(DT_INST_IRQN(idx));							\
+	}											\
+	static const struct gpio_rpi_config gpio_rpi_##idx##_config = {				\
+		.bank_config_func = bank_##idx##_config_func,					\
+		.common =									\
+		{										\
+			.port_pin_mask =							\
+				((~0) & ~(BIT(23) | BIT(24) | BIT(29) | BIT(30) | BIT(31)))	\
```

Comment:
magic const here? can this be defined/documented? isn't `GPIO_PORT_PIN_MASK_FROM_DT_INST` suitable?

Reply:
Indeed it is. Updated to use that macro, added the `ngpios` property in the DTS, and rebased on top of latest `main`.